src/plugins/blocksmith/components/pinterest-pin-block-modal/index.jsx

```diff
@@ -0,0 +1,231 @@
+      touched: false,
+      updatedDate: block.get('updated_date', new Date()),
+    };
+
+    this.handleAddBlock = this.handleAddBlock.bind(this);
+    this.handleChangeCheckbox = this.handleChangeCheckbox.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeTextarea = this.handleChangeTextarea.bind(this);
+    this.handleChangeTime = this.handleChangeTime.bind(this);
+    this.handleEditBlock = this.handleEditBlock.bind(this);
+    this.handleChangeSelect = this.handleChangeSelect.bind(this);
+  }
+
+  componentDidMount() {
+    setTimeout((t) => {
+      t.inputElement.focus();
+    }, 0, this);
+  }
+
+  setBlock() {
+    const { hasUpdatedDate, terse, href, size, updatedDate, aside } = this.state;
+    const { block } = this.props;
+    return block.schema().createMessage()
+      .set('aside', aside)
+      .set('href', href)
+      .set('size', size)
+      .set('terse', terse)
+      .set('updated_date', hasUpdatedDate ? updatedDate : null);
+  }
+
+  handleAddBlock() {
+    const { onAddBlock, toggle } = this.props;
+    onAddBlock(this.setBlock());
+    toggle();
+  }
+
+  handleEditBlock() {
+    const { onEditBlock, toggle } = this.props;
+    onEditBlock(this.setBlock());
+    toggle();
+  }
+
+  handleChangeCheckbox({ target: { checked, id } }) {
+    this.setState({ [id]: checked });
+  }
+
+  handleChangeDate(date) {
+    this.setState(changedDate(date));
+  }
+
+  handleChangeTime({ target: { value: time } }) {
+    this.setState(changedTime(time));
+  }
+
+  handleChangeTextarea(event) {
+    let { errorMsg, href, isValid, size, terse } = this.state;
+    const input = event.target.value;
+    const pinId = getPinterestPinId(input);
+
+    const embed = new DOMParser().parseFromString(input, 'text/html');
+    const anchor = embed.querySelector('a');
+    const pinWidth = anchor ? anchor.dataset.pinWidth : '';
+    const pinTerse = anchor ? anchor.dataset.pinTerse : '';
+
+    if (pinId) {
+      errorMsg = '';
+      isValid = true;
+      href = pinId;
+      size = pinWidth || 'small';
+      terse = !!pinTerse;
+    } else {
+      errorMsg = 'Pinterest pin id, embed url or embed code is invalid';
+      isValid = false;
+    }
+
+    this.setState({
+      errorMsg,
+      href,
+      isValid,
+      size,
+      terse,
+      touched: true,
+    });
+  }
+
+  handleChangeSelect({ value }) {
+    this.setState({ size: value });
+  }
+
+  render() {
+    const {
+      aside,
+      errorMsg,
+      hasUpdatedDate,
+      isValid,
+      href,
+      terse,
+      touched,
+      size,
+      updatedDate,
+    } = this.state;
+    const { isFreshBlock, isOpen, toggle } = this.props;
+    const displayUrl = isValid ? href : '';
+
+    return (
+      <Modal isOpen={isOpen} toggle={toggle}>
+        <ModalHeader toggle={toggle}>{`${isFreshBlock ? 'Add' : 'Update'} Pinterest Pin Block`}</ModalHeader>
+        <ModalBody>
+          <FormGroup>
+            <Label>URL</Label>
```

Comment:
@jalopez13 have the label react to the input state, ref tiktok modal

Reply:
@efjacobson 

Updated to react to isValid. `<Label>{isValid ? 'Pinterest Id' : 'Pinterest Embed Code, URL, or ID'}</Label>`